lib/Controller/ApiTablesController.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace OCA\Tables\Controller;
+
+use Exception;
+use OCA\Tables\Errors\InternalError;
+use OCA\Tables\Errors\NotFoundError;
+use OCA\Tables\Errors\PermissionError;
+use OCA\Tables\ResponseDefinitions;
+use OCA\Tables\Service\TableService;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\IRequest;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @psalm-import-type TablesTable from ResponseDefinitions
+ */
+class ApiTablesController extends MyOCSController {
+	private TableService $service;
+
+	public function __construct(
+		IRequest $request,
+		LoggerInterface $logger,
+		TableService $service,
+		string $userId) {
+		parent::__construct($request, $logger, $userId);
+		$this->service = $service;
+	}
+
+	/**
+	 * [api v2] Returns all Tables
+	 *
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 *
+	 * @return DataResponse<Http::STATUS_OK, TablesTable[], array{}>|DataResponse<Http::STATUS_INTERNAL_SERVER_ERROR, array{message: string}, array{}>
+	 *
+	 * 200: Tables returned
+	 */
+	public function index(): DataResponse {
+		try {
+			return new DataResponse($this->service->formatTables($this->service->findAll($this->userId)));
+		} catch (InternalError|Exception $e) {
+			return $this->handleError($e);
+		}
+	}
+
+	/**
+	 * [api v2] Get a table object
+	 *
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 *
+	 * @param int $id Table ID
+	 * @return DataResponse<Http::STATUS_OK, TablesTable, array{}>|DataResponse<Http::STATUS_FORBIDDEN|Http::STATUS_INTERNAL_SERVER_ERROR|Http::STATUS_NOT_FOUND, array{message: string}, array{}>
+	 *
+	 * 200: Table returned
+	 * 403: No permissions
+	 * 404: Not found
+	 */
+	public function show(int $id): DataResponse {
+		try {
+			return new DataResponse($this->service->find($id)->jsonSerialize());
```

Comment:
```suggestion
			return new DataResponse($this->service->find($id));
```

jsonSerialize should be called automatically once the DataResponse is passed to json_encode, but also fine to keep it explicit here.

Reply:
Yep, this is needed for psalm to understand what is going on